src/native/libs/System.Native/pal_io.c

```diff
@@ -1800,9 +1820,31 @@ uint32_t SystemNative_GetFileSystemType(intptr_t fd)
     uint32_t result = (uint32_t)statfsArgs.f_type;
     return result;
 #endif
+#elif defined(TARGET_HAIKU)
+    struct stat st;
+    int fstatRes;
+    while ((fstatRes = fstat(ToFileDescriptor(fd), &st)) == -1 && errno == EINTR)
+    { }
```

Comment:
My preference would be to use consistent style with the rest of the file.

Reply:
Done.